src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/TransitLayerMapper.java

```diff
@@ -66,18 +81,127 @@ private TransitLayer map() {
     }
 
     /**
-     * Map trip tripPatterns and trips to Raptor classes
+     * Map pre-Raptor TripPatterns and Trips to the corresponding Raptor classes.
      */
-    private HashMap<LocalDate, List<TripPatternForDate>> mapTripPatterns(
-            Map<Stop, Integer> indexByStop
-    ) {
-        return mapTripPatternsByStopDate(
-                mapPatternsByServiceCode(indexByStop,  originalTripPatterns()),
-                mapServiceCodesByLocalDates(graph.getCalendarService(), graph.serviceCodes)
-        );
+    private HashMap<LocalDate, List<TripPatternForDate>> mapTripPatterns (Map<Stop, Integer> indexByStop) {
+        // CalendarService has one main implementation (CalendarServiceImpl) which contains a CalendarServiceData which
+        // can easily supply all of the dates. But it's impossible to actually see those dates without modifying the
+        // interfaces and inheritance. So we have to work around this abstraction and reconstruct the CalendarData.
+        // Note the "multiCalendarServiceImpl" which has docs saying it expects one single CalendarData.
+        // It seems to merge the calendar services from multiple GTFS feeds, but its only documentation says it's a hack.
+
+        // Reconstruct set of all dates where service is defined, keeping track of which services run on which days.
+        CalendarService calendarService = graph.getCalendarService();
+        Multimap<ServiceDate, FeedScopedId> serviceIdsForServiceDate = HashMultimap.create();
+        for (FeedScopedId serviceId : calendarService.getServiceIds()) {
+            Set<ServiceDate> serviceDatesForService = calendarService.getServiceDatesForServiceId(serviceId);
+            for (ServiceDate serviceDate : serviceDatesForService) {
+                serviceIdsForServiceDate.put(serviceDate, serviceId);
+            }
+        }
+
```

Comment:
The creation of the multimap in my mind should be in its own method, I might not care about how it is created - I care about what is needed to create it (input/dependencies) and what it is used for. The same goes for several blocks of code below in this method. This create another problem, the scope of local variables accessible to the blocks below. It is convenient to have access to everything - but it makes it hard to see witch part depend on what. In my eyes god design is about readability, reducing dependencies and encapsulation (here at the level of a code block).

Reply:
This idea applies in several other places as well - this was the main stylistic difference I was referring to. When formulating the overall approach, I made one large method, but there are several subsections that could be pulled out into their own methods. 

When the intent is to show exactly the inputs needed, making the method static can give some hint that we're dealing with a pure function from exactly the given inputs to the needed output.

One hesitation I had about the old code was that it wasn't immediately clear if these smaller methods were lazily creating a since object that could be fetched repeatedly be reuse of the method, or would create a whole new set of instances each time. In practice they were only called once, but you had to understand what the code did already to see that. I could imagine someone refactoring the code and misusing them, by calling the same method N times in a single caller, assuming it was re-fetching the same instance.

Reply:
Added a TODO